src/routes/API.mjs

```diff
@@ -145,8 +146,13 @@ export function clientAuthenticated (target, name, descriptor) {
   const endpoint = descriptor.value
 
   descriptor.value = async function value (...args) {
-    await Authentication.requireClientAuthentication({ connection: args[0] })
-    return endpoint.apply(this, args)
+    const [ctx] = args
+    const client = await Authentication.requireClientAuthentication({ connection: args[0] })
```

Comment:
Couldn't you just pass `ctx` as `connection` directly here instead of accessing args twice?